ExchangeSharp/API/Exchanges/BL3P/ExchangeBL3PAPI.cs

```diff
@@ -138,9 +165,151 @@ Task MessageCallback(IWebSocket _, byte[] msg)
 			);
 		}
 
-		public partial class ExchangeName
+		protected override bool CanMakeAuthenticatedRequest(IReadOnlyDictionary<string, object> payload)
+		{
+			return !(PublicApiKey is null) && !(PrivateApiKey is null);
+		}
+
+		protected override async Task ProcessRequestAsync(IHttpWebRequest request, Dictionary<string, object> payload)
+		{
+			var formData = await request.WritePayloadFormToRequestAsync(payload)
+				.ConfigureAwait(false);
+
+			if (CanMakeAuthenticatedRequest(payload))
+			{
+				request.AddHeader("Rest-Key", PublicApiKey.ToUnsecureString());
+				var signKey = GetSignKey(request, formData);
+				request.AddHeader("Rest-Sign", signKey);
+			}
+		}
+
+		private string GetSignKey(IHttpWebRequest request, string formData)
+		{
+			//TODO: Use csharp8 ranges
```

Comment:
To use `System.Range` we would have to drop `net472` as a a target framework

Reply:
That's not an option unfortunately right now

Reply:
We already support `netstandard2.0` which is compatible [¹] [²] with `net461`. We are not using any APIs that are `net472` specific (AFAIK) :thinking:  

 [¹]: https://github.com/dotnet/standard/blob/master/docs/versions.md
 [²]: https://dotnet.microsoft.com/platform/dotnet-standard

Reply:
Wouldn't that mean that we would have to drop support for netstandard 2.0 as well in order to use System.Range?

Reply:
That is true. One thing that I said that was wrong is that we need to drop `net472` when actually what I should have said is that there's no need to support `net472` when we already support `netstandard20`. With that clarified, I noticed that those types (`Range` and `Index`) are not available in all runtimes (only `netstandard2.1` and `netcoreapp30`) and in that case we could use a package that ports those specific APIs to `netstandard20` like this one https://github.com/bgrainger/IndexRange

Reply:
Maybe we could create an issue out of this conversation :thinking:

Reply:
> we could use a package that ports those specific APIs to `netstandard20` like this one https://github.com/bgrainger/IndexRange

Yes that's a good way to keep netstandard20 and start using the new APIs




> Maybe we could create an issue out of this conversation 🤔

Yes feel free